### ISSUE TITLE
AbstractFunctionDefinition.CommandResultType Single explodes if I inherit an interface from ICommand<T>

### ISSUE THREAD

**Issue by abezverkov** [JamesRandall/FunctionMonkey#27]
For example, the following code throws an exception at build in FM.Compiler.
```
public interface ICustomCommand<T>: ICommand<T> { } // some custom stuff
public class MyFooCustomCommand: ICustomCommand<Foo>
```
The code throwing the exception is:
```
Type genericCommandInterface = CommandType.GetInterfaces()
.SingleOrDefault(x => x.IsGenericType && commandInterface.IsAssignableFrom(x));
```
Because the `GetInterfaces()` is returning `ICommand<Foo>` and `ICustomCommand<Foo>`.

It seems you are looking for just `ICommand<Foo>` in this case.  Since this is at compile, you could narrow it down by adding a check on the interfaces for `x`.   `ICommand<Foo>` will only have the one so ` && x.GetInterfaces().Count() ==1` would work (since you are already checking IsAssignableFrom(x)).

Thoughts?

**Comment by JamesRandall** [comment]
Thanks for reporting this - sorry it took me so long to get to.

Your proposed solution makes sense - I'll look to get that implemented in the next release. Thank you!

**Comment by JamesRandall** [comment]
Resolved in issue v0.23.0
